scripts/conflicts/diff.js

```diff
@@ -20,37 +26,90 @@ if (!server || argv.h || argv.help) {
 
 console.log(`Generating diffs for conflicts on ${server}`);
 
+
 const DB = new PouchDB(server);
 
-DB.query('medic-conflicts/conflicts')
+
+const writeToFile = function(path, content) {
+  const writeStream = fs.createWriteStream(path);
+  writeStream.write(JSON.stringify(content, null, 3));
+  writeStream.close();
+};
+
+DB.query('medic-conflicts/conflicts', {reduce:false})
   .then(conflicts => {
     console.log(`Found ${conflicts.rows.length} conflicts`);
 
+    const mainConflictFilesDirectory = './doc-conflicts';
     let ps = Promise.resolve();
+    if (!fs.existsSync(mainConflictFilesDirectory)){
+      try {
+        fs.mkdirSync(mainConflictFilesDirectory);
+      }catch(err){
+        console.log(`Error while creating main conflict folder: ${err}`);
+      }
+    }
     conflicts.rows.forEach(r => {
       ps = ps.then(() => {
-        // console.log(r);
-        if (r.key.length !== 1) {
-          console.log(`${r.id} has ${r.key.length} conflicts, we only want to deal with 1, skipping...`);
-        } else {
-          const docId = r.id;
-          const conflictRev = r.key[0];
-          return DB.get(docId, { open_revs: 'all' })
-            .then(results => {
-              // console.log(results);
-              // Make sure mine and theirs are the right way around
-              const mine = results.find(r => r.ok._rev !== conflictRev).ok;
-              const theirs = results.find(r => r.ok._rev === conflictRev).ok;
-
-              console.log('=====================');
-              console.log(r.id);
-              console.log('=====================');
-              console.log(jsondiff.diffString(mine, theirs));
-              console.log('\n\n');
-            });
-        }
+        const docId = r.id;
+        const arrayRevsIdThatShouldNotBeThere = r.key;
+        return DB.get(docId, { open_revs: 'all' })
+          .then(results => {
+
+            // Get the main_doc. Maindoc is the doc that isn't in the conflict list
+            const main_doc = results.filter(r => !arrayRevsIdThatShouldNotBeThere.includes(r.ok._rev))[0].ok; 
+
+            //Now we are going to put each doc, its conflicts and thier diff 
+            //in one file for manual inspection if required.
+            //The doc id might have characters that cannot be used to create folders.
+            let conflictDirectoryName = r.id;
+            if(r.id.includes('target')){
+              conflictDirectoryName = 'target_' + r.id.split(':')[0];
+            } else if (r.id.includes('task')){ //If doc is task add ask and part of doc id
+              conflictDirectoryName = 'task_' + r.id.split(':')[1];
```

Comment:
As above, an example id for a task document is: `task~org.couchdb.user:adventives~4b710ca5-6c86-43a7-97ef-797ec5435dbc~pregnancy-danger-sign-follow-up~a random task4~1643343620261`. 
Is it super valuable to strip out `task~org.couchdb.user:` ?